common/PerfStressShared/PerfStressUtilities.cs

```diff
@@ -0,0 +1,133 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using CommandLine;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Reflection;
+using System.Reflection.Emit;
+using System.Text.RegularExpressions;
+using System.Threading;
+
+namespace Azure.Test.PerfStress
+{
+    internal static class PerfStressUtilities
+    {
+        private static readonly Lazy<Parser> _parser = new Lazy<Parser>(() => new Parser(settings =>
+        {
+            settings.CaseSensitive = false;
+            settings.HelpWriter = Console.Error;
+        }));
+
+        internal static Parser Parser => _parser.Value;
+
+        internal static bool ContainsOperationCanceledException(Exception e)
+        {
+            if (e is OperationCanceledException)
+            {
+                return true;
+            }
+            else if (e.InnerException != null)
+            {
+                return ContainsOperationCanceledException(e.InnerException);
+            }
+            else
+            {
+                return false;
+            }
+        }
+
+        // Dynamically create option types with a "Verb" attribute
+        internal static Type[] GetOptionTypes(IEnumerable<Type> testTypes)
+        {
+            var optionTypes = new List<Type>();
+
+            var ab = AssemblyBuilder.DefineDynamicAssembly(new AssemblyName("Options"), AssemblyBuilderAccess.Run);
+            var mb = ab.DefineDynamicModule("Options");
+
+            foreach (var t in testTypes)
+            {
+                var baseOptionsType = t.GetConstructors().First().GetParameters()[0].ParameterType;
```

Comment:
`First()` will throw a cryptic error at this level. Maybe `FirstOrDefault` and check, erring or throwing with a better message at least? Seems also you should enumerate constructors. There's a lot of assumptions going on here.

Reply:
The test framework assumes test classes have a single ctor which accepts the `Options` object.  I will improve the logic to handle multiple ctors (as long as one matches the expected ctor), and throw a nice error if none match.

#16971